### PR TITLE
[7/N] Fixes clang-tidy warnings in c10/{core,util}/*.h

### PR DESCRIPTION
**Pull request by cyyever**
This PR continues to fix clang-tidy warnings for headers in c10/core and c10/util.

### CODE REVIEW

File: c10/core/Stream.h [hunk 1]
```diff
@@ -1,6 +1,13 @@
 #pragma once
 
 #include <c10/core/Device.h>
+#include <c10/core/DeviceType.h>
+#include <c10/macros/Export.h>
+#include <c10/util/Exception.h>
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+#include <ostream>
```
Comment (Skylion007) [inline c10/core/Stream.h:10]: Do we need ostream for all these headers? Or would iosfwd be sufficient?

File: c10/util/SmallVector.h [hunk 2]
```diff
@@ -376,9 +373,9 @@ class SmallVectorTemplateCommon
 /// note
 template <
     typename T,
-    bool = (std::is_trivially_copy_constructible<T>::value) &&
-        (std::is_trivially_move_constructible<T>::value) &&
-        std::is_trivially_destructible<T>::value>
+    bool = (std::is_trivially_copy_constructible_v<T>)&&(
+        std::is_trivially_move_constructible_v<
+            T>)&&std::is_trivially_destructible_v<T>>
```
Comment (Skylion007) [inline c10/util/SmallVector.h:378]: Was clang-format not run on this file? Seems like odd style not have spaces around &&

**Review by Skylion007 — COMMENTED** [review]
Some nits.

File: c10/util/SmallVector.h (hunk 2 above)
Comment (cyyever) [reply c10/util/SmallVector.h:378]: The current lower version of clang-format encourages this style...

File: c10/core/Stream.h (hunk 1 above)
Comment (cyyever) [reply c10/core/Stream.h:10]: ostream functions are used in the header.

**Comment by cyyever** [comment]
@Skylion007 We need to update clang-format to deal with bad formatting

**Comment by Skylion007** [comment]
@cyyever We should ping @malfet to see if we can update clang-format to fix this then.

**Comment by malfet** [comment]
Updating clang-format (unlike clang-tidy) should be a no brainer. Give me an hour...

**Review by malfet — APPROVED** [review]

File: c10/util/SmallVector.h (hunk 2 above)
Comment (malfet) [reply c10/util/SmallVector.h:378]: https://github.com/pytorch/pytorch/pull/116002 <- hope this will help

**Pull request closed without merging**